src/Head2Head.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from "react";
+import {
+	getRaces,
+	getSlugData,
+	numPagesForGame,
+	RACETIME,
+	SlugData,
+} from "./Loader";
+import FormattedTable from "./FmtTable";
+import CategoryLauncher from "./CategoryLauncher";
+import "./Head2Head.css";
+import MUTable, { MUEntrant, MUMapCtor } from "./MUTable";
+
+// Used to cancel the async function that loads races if a new slug is set.
+let token: Symbol;
+
+function Head2Head() {
+	const [muTable, setMuTable] = useState(new MUTable());
```

Comment:
React really likes immutable data! By centering your UI around a mutable Object in the sense of OOP you're making it a bit difficult to correctly integrate the code with other React features like memoization or effects. Ask yourself if you really need a singleton object - I think you don't, especially considering the fact that you always call `generateSortedTable` to generate a large amount of data more or less from scratch.

You don't have to give up on the MUTable class entirely! Perhaps it might make more sense to temporarily construct a MUTable class in a `useMemo` call where you're computing derived data without letting this MUTable escape to React?